### ISSUE TITLE
Host a Nextcloud instance for club members

### ISSUE THREAD

**Issue by jflory7** [RITlug/tasks#14]
From a short discussion in IRC with @Serubin about hosting a [Nextcloud](https://nextcloud.com) instance for club members. More details to come soon.

**Comment by justwheel** [comment]
@ct-martin completed a proof-of-concept over this in the winter break. We'll table this for discussion in the next two weeks.

**Comment by ct-martin** [comment]
As per eboard meetings:

Ubuntu 16.04 LTS
2 core, 4GB RAM
40GB Main OS Disk
100GB Data/Storage Disk

**Comment by justwheel** [comment]
@ct-martin I'm okay with what works best for the install from our resource pool, but this is blocked by policy decisions. Our requirements may change once we know our use case and how we want to offer this to the club.

**Comment by Tjzabel** [comment]
@jflory7 understandable. We spoke to @Serubin about this, and he gave the okay for the allocated resources. What exactly is this blocked on?

**Comment by justwheel** [comment]
@Tjzabel I created [this milestone](https://github.com/RITlug/tasks/milestone/4 "NextCloud now offered to club") to help visualize the work left to completion.

We can move forward on provisioning the machine (#29) and doing a base install (#30). Customizations and configuration changes (#31) is blocked by defining policy for the server:

1. How do we determine eligibility criteria to receive a NextCloud account?
2. Does this require us to reconsider how we define membership?
3. How do we define access to potentially private data as RITlug's eboard?

And possibly other questions.

**Comment by Serubin** [comment]
@Tjzabel @jflory7 See https://github.com/RITlug/tasks/issues/29

**Comment by ct-martin** [comment]
RITlug has other tasks of higher priority right now so I have put the Nextcloud instance in maintenance mode and moved this issue to the backlog

**Comment by justwheel** [comment]
@ct-martin None of us have discussed what current priorities are. It would be helpful to get us on the same page so we can understand your thinking here.

**Comment by Tjzabel** [comment]
@jwflory @ct-martin I am wondering if having a NextCloud instance is a worthy project to undertake at this point in time. 

For me personally, I only see having NextCloud as a great tool if we were to replace our Google Drive with it. While having our own NextCloud instance for our members would be a nice-to-have, I'm not sure if it's worth the technical debt we may accrue. 

Thoughts?

**Comment by ct-martin** [comment]
Having an open alternative to our Google Drive would be nice, as well as a
place to store backups of photos from events. One thing for policy I would
put forward is that we don't give members personal space, and allocate
shared group folders for different SIGs. Most of the setup is done, but
blocking on a couple infra things.

--
Christian Martin (Phone)

**Comment by Tjzabel** [comment]
@ct-martin hm, interesting. I would like to talk about this before we get much further in the physical infra work. 
If you wouldn't mind creating an outline or a proof-of-concept for what you think the use-case of NextCloud is, I would love to see that. I would like to know more about what you mean by having shared folders instead of having personal space.

**Comment by Tjzabel** [comment]
### Update
This issue is blocked on me resetting the NextCloud VM root password for @ct-martin.

#### Second Update
I need to stop writing updates and actually *reset the password*

**Comment by Tjzabel** [comment]
### Update
Finally reset the password! We're now unblocked with this issue.

Me, @ct-martin and @jrtechs have accounts on the nextcloud box with sudo access.

**Comment by ct-martin** [comment]
For clarification, #31 is blocked on policy discussion. Eboard members were sent logins over Slack
